exercises/satellite/canonical-data.json

```
@@ -74,6 +74,42 @@
       "expected": {
         "error": "traversals must contain unique items"
       }
+    },
+    {
+      "uuid": "d232a4e5-622e-4aaa-965e-244ddcdcdb96",
+      "description": "Complex tree with many integer values",
+      "property": "treeFromTraversals",
+      "input": {
+        "preorder": ["2", "1", "3", "6", "5", "7"],
+        "inorder": ["1", "2", "3", "5", "6", "7"]
+      },
+      "expected": {
+        "v": "4",
+        "l": {
+          "v": "2",
+          "l": { "v": "1", "l": {}, "r": {} },
+          "r": { "v": "3", "l": {}, "r": {} }
+        },
+        "r": {
+          "v": "6",
+          "l": { "v": "5", "l": {}, "r": {} },
+          "r": { "v": "7", "l": {}, "r": {} }
+        }
+      }
+    },
+    {
+      "uuid": "156a2dd8-a2c7-4a07-b9fe-ab599956d067",
+      "description": "Simple tree with integer values",
+      "property": "treeFromTraversals",
+      "input": {
+        "preorder": ["2", "1"],
+        "inorder": ["1", "2"]
+      },
+      "expected": {
+        "v": "2",
+        "l": "1",
```

```suggestion
        "l": { "v": "1", "l": {}, "r": {} },
```

a tree node found at `"l"` cannot be a string; it has to be an object with a `"v"`, an `"l"`, and an `"r"` (like all other cases in this file)